### PR TITLE
Service, context, updater stubbing

### PR DESCRIPTION
**Pull request by gabriel**
This PR removes a lot of v1 code, updates some interfaces for v2.

- Context fixes
- Service starts up context (and starts checker)
- Update checker simplified for v2
- Remove singleflight
- Main updater code removed (stubbed out)

### CODE REVIEW

**Comment by gabriel** [comment]
@keybase/updater-hackers

**Comment by chrisnojima** [comment]
@keybase/react-hackers bump
@jzila

File: updater.go [hunk 1]
```diff
@@ -3,833 +3,55 @@
...
-	switch err.(type) {
-	case CanceledError:
-		// Canceled errors aren't really errors
-		return keybase1.StatusOK(err.Error())
-	default:
-		return keybase1.FromError(err).Status()
-	}
+// Update performs an update
+func (u *Updater) Update(ctx Context) (*Update, error) {
+	// TODO
```
Comment (chrisnojima) [inline updater.go]: ?

**Comment by chrisnojima** [comment]
this method of merging into this brach and keeping the old code and overwriting it makes it logically hard to understand 'where we're at' in the process and what we'll end up with. unclear if there's something we can do at this point but its hard to feel confident in the PRs

**Comment by chrisnojima** [comment]
@jzila can you take a look

File: keybase/context.go [hunk 2]
```diff
@@ -23,48 +22,71 @@ var validCodeSigningKIDs = map[string]bool{
...
+}
+
 // UpdateOptions returns update options
 func (c *context) UpdateOptions() updater.UpdateOptions {
 	return c.config.updaterOptions()
 }
 
+// GetUpdateUI returns Update UI
+func (c *context) GetUpdateUI() (updater.UpdateUI, error) {
+	return c, nil
```
Comment (jzila) [inline keybase/context.go]: is this just to export the `UpdateUI` portions of this context?

File: keybase/context.go [hunk 3]
```diff
@@ -23,48 +22,71 @@ var validCodeSigningKIDs = map[string]bool{
...
 	log logging.Logger
 }
 
-func newContext(cfg *config, log logging.Logger) *context {
-	ctx := context{}
-	ctx.config = cfg
-	ctx.log = log
+func newContext(upd *updater.Updater, cfg *config, log logging.Logger) *context {
+	ctx := context{
+		updater: upd,
```
Comment (jzila) [inline keybase/context.go]: I can't find where this is used

File: saltpack_test.go [hunk 4]
```diff
@@ -58,28 +67,20 @@ func TestSaltpackVerifyBadValidIDs(t *testing.T) {
 
 	reader := bytes.NewReader([]byte(message1))
 	err := SaltpackVerifyDetached(reader, signature1, badCodeSigningKIDs, log)
+	assert.Error(t, err, "Should have failed verify")
 	t.Logf("Error: %s", err)
```
Comment (jzila) [inline saltpack_test.go]: Do we want to log nil errors?

**Comment by jzila** [comment]
Mostly looks good. Regarding the tests, it seems like you're using `testify/assert` and logging some of the errors, whereas the previous code would fatal out if a test condition failed. Do you want to use `testify/require` to replicate that?

LGTM after nits/questions.

File: keybase/context.go (hunk 2 above)
Comment (gabriel) [inline keybase/context.go]: yeah, the context object implements the UpdateUI interface

File: keybase/context.go (hunk 3 above)
Comment (gabriel) [inline keybase/context.go]: Yeah it's not used, will remove

File: saltpack_test.go (hunk 4 above)
Comment (gabriel) [inline saltpack_test.go]: Nope... did a pass where I changed some asserts to require

**Comment by gabriel** [comment]
Yep, did a pass and changed some asserts to require.

**Comment by gabriel** [comment]
@chrisnojima Yeah this PR was a bit messy cause I was removing code and re-adding stubs and a couple interfaces changed which affected different packages. I should have removed code in a separate PR and then done smaller PRs for the changes here, which became more obvious after I had made a bunch of changes in the same branch :(

**Pull request merged**